syntax = "proto3";

// The greeting service definition.
service Greeter {
  // Sends a greeting
  rpc SayHello (HelloRequest) returns (HelloReply) {}
  rpc ListNames (HelloRequest) returns (stream HelloReply) {}
  rpc HelloMany (stream HelloRequest) returns (HelloReply) {}
  rpc HelloChat (stream HelloRequest) returns (stream HelloReply) {}
}

// The request message containing the user's name.
message HelloRequest {
  string name = 1;
}

// The response message containing the greetings
message HelloReply {
  string message = 1;
}

